jicoco/src/main/java/org/jitsi/xmpp/mucclient/MucClient.java

```diff
@@ -300,7 +306,27 @@ public void reconnectionFailed(Exception e)
         {
             logger.debug("About to connect and login.");
         }
-        xmppConnection.connect().login();
+
+        try
+        {
+            xmppConnection.connect();
+        }
+        catch(Throwable t)
+        {
+            logger.warn(MucClient.this + " error connecting will retry:", t);
+
+            ScheduledExecutorService executorService
```

Comment:
I see there's an executor instance managed by the `MucClientManager` already. Maybe just pass it as an argument to this method instead of trying to manage another one here?

Reply:
I think this line  needs to be changed as well: https://github.com/jitsi/jicoco/blob/master/jicoco/src/main/java/org/jitsi/xmpp/mucclient/MucClientManager.java#L149
it would call `initializeConnectAndJoin` by passing  the executor.

Reply:
Ok, will try that one. Just having problems with IDE since we added the kotlin stuff in here :(

Reply:
Those are different executors... So we need to change the API in one of the places in order to make this change ...